### PR TITLE
Add sea ice production/melting analysis tasks

### LINKED ISSUE #216: Add analysis of sea ice formation and melt rates
Assuming rates of sea ice formation and melting are available in the timeSeriesStats AM, an analysis task that compares these against SOSE and possibly satellite-derived formation rates (e.g. Tamura 2008) would be useful.

### PR DESCRIPTION
**Pull request by darincomeau**
Adds analysis task for climatology maps of sea ice total production and freezing, with comparisons to observations based on [Haumann et al 2016](https://www.nature.com/articles/nature19101). Note observations are only available for annual climatology, Southern Hemisphere.

closes #216

### CODE REVIEW

**Comment by darincomeau** [comment]
I haven't gotten the observations to plot yet, but here's what I have so far in terms of production/melting plots:

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/20220901.DTESTM.T62_EC30to60E2r2.QCtest.anvil/yrs1-5/sea_ice/index.html

That directory will probably be wiped on my next run that errors out, so I'll post some plots here:
imageimage

**Comment by darincomeau** [comment]
FYI @xylar the obs data is now getting remapped, and the error I'm seeing is at the stage of plotting the remapped obs data, so I think we're almost there (thanks for the help today!).

File: mpas_analysis/sea_ice/climatology_map_melting.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+        """
+        # Authors
+        # -------
+        # Darin Comeau, Xylar Asay-Davis
+
+        dsObs = xr.open_dataset(fileName)
+        dsObs = dsObs.rename({'melting': 'seaIceMelting', 'time': 'Time'})
+        # dsObs.coords['month'] = dsObs['Time.month']
+        # dsObs.coords['year'] = dsObs['Time.year']
+        dsObs = dsObs.transpose('Time', 'y', 'x')
```
Comment (xylar) [inline mpas_analysis/sea_ice/climatology_map_melting.py:328]: This seems to be working for me. I think you want `Time` in there if you're going to compute a climatology and you don't if the climatology has already been computed (so `year` and `months` also aren't there).
```suggestion
        dsObs = dsObs.isel(time=0)
        dsObs = dsObs.rename({'melting': 'seaIceMelting'})
        dsObs = dsObs.transpose('y', 'x')
```

File: mpas_analysis/sea_ice/climatology_map_production.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        """
+        # Authors
+        # -------
+        # Darin Comeau, Xylar Asay-Davis
+
+        dsObs = xr.open_dataset(fileName)
+        dsObs = dsObs.rename({'freezing': 'seaIceProduction', 'time': 'Time'})
+        # dsObs.coords['month'] = dsObs['Time.month']
+        # dsObs.coords['year'] = dsObs['Time.year']
+        dsObs = dsObs.transpose('Time', 'y', 'x')
```
Comment (xylar) [inline mpas_analysis/sea_ice/climatology_map_production.py:328]:
```suggestion
        dsObs = dsObs.isel(time=0)
        dsObs = dsObs.rename({'freezing': 'seaIceProduction'})
        dsObs = dsObs.transpose('y', 'x')
```

**Comment by xylar** [comment]
Hmm, I tried my suggestions above but the obs came out blank so there's debugging still to do:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis_testing/sea_ice_prod_melt/20220901.DTESTM.T62_EC30to60E2r2.QCtest.anvil/yrs1-5/sea_ice/index.html

**Comment by darincomeau** [comment]
I ran this on a *real* run, and tweaked some of the plotting config options as a result:

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/20220808.v2.cryo.Bcase.submesoscale.ne30pg2_SOwISC12to60E2r4.chrysalis/yrs91-100/sea_ice/index.html

I'll talk briefly about this at tomorrow's check-in, but I think this is pretty much ready for review.

**Comment by darincomeau** [comment]
Closes https://github.com/MPAS-Dev/MPAS-Analysis/issues/216

**Comment by xylar** [comment]
> Closes https://github.com/MPAS-Dev/MPAS-Analysis/issues/216

@darincomeau, this needs to go in the description, not a comment.  I'll take care of it.

**Review by xylar — APPROVED** [review]
This looks awesome and I'm excited to have it!

My one suggestion would be to go ahead and put everything in the same gallery, rather than having one with obs and one without.  It's a little clunky either way since the thumbnails are different sizes but it makes comparing between seasons and annual a *little* bit easier:

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xylar/analysis_testing/sea_ice_prod_melt/v2.cryo.Bcase.submesoscale.ne30pg2_SOwISC12to60E2r4.chrysalis/clim_0001-0020_ts_0001-0020/sea_ice/index.html

I had mentioned that the filenames for the remapped observations looked weird to me.  That's nothing you did -- MPAS-Analysis just gives remapped obs weird filenames, I guess.  They contain both the original and the remapped grid resolutions, which was a surprise to me.  Anyway, nothing we need to do anything about here.

**Comment by xylar** [comment]
@akturner and @milenaveneziani, please have a look when you can.  It would be great to get this in!

**Comment by darincomeau** [comment]
@xylar good suggestion, I've implemented it. I've also made the observations name change to AnIceFlux. Output now:

https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.dcomeau/20220808.v2.cryo.Bcase.submesoscale.ne30pg2_SOwISC12to60E2r4.chrysalis/yrs91-100/sea_ice/index.html

One other thing I meant to solicit opinion on is whether this belongs in `polar_regions.cfg`. Now it's part of the regular output - I'll leave it there unless there's a stated preference to move it to polar_regions.

**Comment by xylar** [comment]
> One other thing I meant to solicit opinion on is whether this belongs in polar_regions.cfg. Now it's part of the regular output - I'll leave it there unless there's a stated preference to move it to polar_regions.

I think this should be on regardless.  I think WC would be interested in it, not just cryosphere.

**Comment by milenaveneziani** [comment]
This looks great!
One only visual suggestion from me: could you please revert the colorer colors for the the Sea Ice Production bias in the SH? (blue for negative bias and red for positive bias)

**Comment by darincomeau** [comment]
Thanks @milenaveneziani ! I've reverted that change per your suggestion, and reran my test analysis linked above to reflect that.

**Review by milenaveneziani — APPROVED** [review]
Thanks Darin! this is good to go from my side.

**Comment by xylar** [comment]
@akturner, just a reminder that we're waiting on your review of this one.

**Comment by xylar** [comment]
I'm taking @akturner off this for now.  Seems like 2 reviews is probably enough.

**Comment by xylar** [comment]
@darincomeau, thanks for the hard work on this!

**Comment by xylar** [comment]
Obviously, since I merged this, it'll need to be its own PR.

**Comment by darincomeau** [comment]
@xylar - sure, I'll put together those docs and open a PR.

Also, this PR's functionality is broken on `develop`. @cbegeman has the fix in https://github.com/MPAS-Dev/MPAS-Analysis/pull/917/commits/dd29f0cf995f4fb76f7275fc293463ceaae12e64 (thanks for catching that!).

I tracked it down to the time gap between when I created this branch off develop and when https://github.com/MPAS-Dev/MPAS-Analysis/pull/913 was merged a day later. I suppose I should have rebased, which would have caught this.

Anyways, happy to leave the fix in https://github.com/MPAS-Dev/MPAS-Analysis/pull/917, or if it makes sense to do in the doc PR, can do that as well.

**Comment by xylar** [comment]
@darincomeau, I saw this too late but I think it's fine that the fix goes in in @cbegeman's PR even though it's unrelated.  And sorry for missing that in my testing, too.

**Pull request merged**